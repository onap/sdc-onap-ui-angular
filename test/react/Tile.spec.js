import React from 'react';
import Tile from '../../src/react/Tile.js';
import TileInfo from '../../src/react/TileInfo.js';
import TileInfoLine from '../../src/react/TileInfoLine.js';
import TileFooter from '../../src/react/TileFooter.js';
import TileFooterCell from '../../src/react/TileFooterCell.js';

import renderer from 'react-test-renderer';

describe('Tile', () => {
	test('Empty tile', () => {
		const tile = renderer.create(<Tile />).toJSON();
		expect(tile).toMatchSnapshot();
	});

	test('Tile with props', () => {
		const tile = renderer.create(<Tile headerText='header' headerColor='blue' iconName='vlm' iconColor='blue' />).toJSON();
		expect(tile).toMatchSnapshot();
	});

	test('Tile with content info', () => {
		const tile = renderer.create(<Tile><TileInfo align='center'><TileInfoLine type='title'>Info</TileInfoLine></TileInfo></Tile>).toJSON();
		expect(tile).toMatchSnapshot();
	});

	test('Tile with footer', () => {
		const tile = renderer.create(<Tile><TileFooter align='center'><TileFooterCell>Footer</TileFooterCell></TileFooter></Tile>).toJSON();
		expect(tile).toMatchSnapshot();
	});
});
