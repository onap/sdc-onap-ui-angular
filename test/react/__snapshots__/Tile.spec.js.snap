// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Empty tile 1`] = `
<div
  className="sdc-tile "
  data-test-id={undefined}
  onClick={undefined}
>
  <div
    className="sdc-tile-header "
  />
  <div
    className="sdc-tile-content"
  >
    <div
      className="sdc-tile-content-icon "
    />
  </div>
</div>
`;

exports[`Tile Tile with content info 1`] = `
<div
  className="sdc-tile "
  data-test-id={undefined}
  onClick={undefined}
>
  <div
    className="sdc-tile-header "
  />
  <div
    className="sdc-tile-content"
  >
    <div
      className="sdc-tile-content-icon "
    />
    <div
      className="sdc-tile-content-info centered"
    >
      <div
        className="sdc-tile-info-line title "
        data-test-id={undefined}
      >
        Info
      </div>
    </div>
  </div>
</div>
`;

exports[`Tile Tile with footer 1`] = `
<div
  className="sdc-tile "
  data-test-id={undefined}
  onClick={undefined}
>
  <div
    className="sdc-tile-header "
  />
  <div
    className="sdc-tile-content"
  >
    <div
      className="sdc-tile-content-icon "
    />
  </div>
  <div
    className="sdc-tile-footer centered"
  >
    <span
      className="sdc-tile-footer-cell "
      data-test-id={undefined}
    >
      Footer
    </span>
  </div>
</div>
`;

exports[`Tile Tile with props 1`] = `
<div
  className="sdc-tile "
  data-test-id={undefined}
  onClick={undefined}
>
  <div
    className="sdc-tile-header blue"
  >
    header
  </div>
  <div
    className="sdc-tile-content"
  >
    <div
      className="sdc-tile-content-icon blue"
    >
      <div
        className="svg-icon-wrapper     bottom"
        disabled={undefined}
        onClick={undefined}
      >
        <svg />
        
      </div>
    </div>
  </div>
</div>
`;
